### PR TITLE
Hsla scalar type

### PR DESCRIPTION
**Pull request by wonbyte**
Add `Hsla` scalar type to `Hotchocolate.Types.Scalars`

Addresses #3085

### CODE REVIEW

File: src/HotChocolate/Core/src/Types.Scalars/HslaColorCodeType.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The `HslaColorCode` scalar type represents a valid a CSS HSLA color as defined
+    /// here https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#hsl_colors.
+    /// </summary>
+    public class HslaColorCodeType : StringType
+    {
+        private static readonly string _validationPattern =
+            ScalarResources.HslColorCodeType_ValidationPattern;
```
Comment (wonbyte) [inline src/HotChocolate/Core/src/Types.Scalars/HslaColorCodeType.cs:13]: We can just leverage the HSL regex as it covers both

Comment (PascalSenn) [reply src/HotChocolate/Core/src/Types.Scalars/HslaColorCodeType.cs:13]: Ah, in this case we can just extend the type I think
```
public class HslaColorCodeType : HslColorCodeType 
{
    ..ctors
}
```

**Comment by wonbyte** [comment]
worth a shot!

File: src/HotChocolate/Core/src/Types.Scalars/HexType.cs [hunk 2]
```diff
@@ -4,30 +4,30 @@
 namespace HotChocolate.Types.Scalars
 {
     /// <summary>
-    /// The HexColorCode scalar type represents a valid hex color code.
+    /// The Hex scalar type represents a valid hex color code.
     /// </summary>
-    public class HexColorCodeType : StringType
+    public class HexType : StringType
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/HexType.cs:9]: I think this is to generic. 
`Hex` could also be something other than a color code i think

File: src/HotChocolate/Core/src/Types.Scalars/HslType.cs [hunk 3]
```diff
@@ -4,31 +4,31 @@
 namespace HotChocolate.Types.Scalars
 {
     /// <summary>
-    /// The `HslColorCode` scalar type represents a valid a CSS HSL color as defined
+    /// The `Hsl` scalar type represents a valid a Css Hsl color as defined
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/HslType.cs:7]: In comments it is valid to use captial case, just the dotnet types have this convention
```suggestion
    /// The `Hsl` scalar type represents a valid a CSS HSL color as defined
```

File: src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx [hunk 4]
```diff
@@ -72,38 +72,38 @@
...
   </data>
-  <data name="HslColorCodeType_Description" xml:space="preserve">
-    <value>The HslColorCode scalar type represents a a valid a CSS HSL color defined here https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#hsl_colors.</value>
+  <data name="HslType_Description" xml:space="preserve">
+    <value>The Hsl scalar type represents a a valid a CSS HSL color defined here https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#hsl_colors.</value>
   </data>
-  <data name="HslColorCodeType_IsInvalid_ParseLiteral" xml:space="preserve">
-    <value>HslColorCodeType cannot parse the provided literal. The provided value is not a valid CSS HSL color code.</value>
+  <data name="HslType_IsInvalid_ParseLiteral" xml:space="preserve">
+    <value>HslType cannot parse the provided literal. The provided value is not a valid Css Hsl color code.</value>
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx:148]:
```suggestion
    <value>HslType cannot parse the provided literal. The provided value is not a valid CSS HSL color code.</value>
```

File: src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx [hunk 5]
```diff
@@ -72,38 +72,38 @@
...
   </data>
   <data name="HslColorCodeType_IsInvalid_ParseValue" xml:space="preserve">
-    <value>HslColorCodeType cannot parse the provided value. The provided value is not a valid CSS HSL color code.</value>
+    <value>HslType cannot parse the provided value. The provided value is not a valid Css Hsl color code.</value>
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx:151]:
```suggestion
    <value>HslType cannot parse the provided value. The provided value is not a valid CSS HSL color code.</value>
```

File: src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx [hunk 6]
```diff
@@ -72,38 +72,38 @@
...
   </data>
-  <data name="HslaColorCodeType_Description" xml:space="preserve">
-    <value>The HslaColorCode scalar type represents a a valid a CSS HSLA color as defined here https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#hsl_colors.</value>
+  <data name="HslaType_Description" xml:space="preserve">
+    <value>The Hsla scalar type represents a a valid a CSS HSLA color as defined here https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#hsl_colors.</value>
   </data>
-  <data name="HslaColorCodeType_IsInvalid_ParseLiteral" xml:space="preserve">
-    <value>HslaColorCodeType cannot parse the provided literal. The provided value is not a valid CSS HSLA color code.</value>
+  <data name="HslaType_IsInvalid_ParseLiteral" xml:space="preserve">
+    <value>HslaType cannot parse the provided literal. The provided value is not a valid Css Hsla color code.</value>
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx:160]:
```suggestion
    <value>HslaType cannot parse the provided literal. The provided value is not a valid CSS HSLA color code.</value>
```

File: src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx [hunk 7]
```diff
@@ -72,38 +72,38 @@
...
   </data>
-  <data name="HslaColorCodeType_IsInvalid_ParseValue" xml:space="preserve">
-    <value>HslaColorCodeType cannot parse the provided value. The provided value is not a valid CSS HSLA color code.</value>
+  <data name="HslaType_IsInvalid_ParseValue" xml:space="preserve">
+    <value>HslaType cannot parse the provided value. The provided value is not a valid Css Hsla color code.</value>
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/ScalarResources.resx:163]:
```suggestion
    <value>HslaType cannot parse the provided value. The provided value is not a valid CSS HSLA color code.</value>
```

File: src/HotChocolate/Core/src/Types.Scalars/HslaType.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The `Hsla` scalar type represents a valid a Css Hsla color as defined
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/HslaType.cs:7]:
```suggestion
    /// The `Hsla` scalar type represents a valid a CSS HSLA color as defined
```

**Pull request closed without merging**